text/stacks/0004-jammy-jellyfish.md

```diff
@@ -0,0 +1,113 @@
+# Stacks based on Ubuntu 2022.04: Jammy Jellyfish
+
+## Summary
+
+A set of stacks based on the Ubuntu 2022.04 LTS (Jammy Jellyfish) release base
+image should be developed, released, and maintained by the Stacks subteam. Like
+the existing Bionic stacks, these new stacks should come in "full", "base", and
+"tiny" variants with similar, if not identical, sets of packages pre-installed.
+
+## Motivation
+
+Ubuntu provides a long-term-support (LTS) release every 2 years in April. These
+releases are supported by Canonical for 5 years. The current Bionic base image,
+first released in April 2018, will therefore be going out of support at the end
+of March 2023. In order to ensure that Paketo continues to provide a supported
+Ubuntu-based stack for our users, we should produce a new set of stack images,
+based on Jammy, alongside the current Bionic stacks.
+
+## Detailed Explanation
+
+### Variants
+
+The Jammy stack will be delivered in 3 variants that align to those already
+offered in the Bionic stack. The variants will be called `base`, `full`, and
+`tiny` just as they are in the Bionic stack. Each should be developed to offer
+roughly the same set of OS-level package support as their Bionic equivalent.
+
+### Stack IDs
+
+Each of these stacks will declare their stack ID as `io.paketo.stack.jammy`.
+This choice differs from Bionic in that we will **NOT** differentiate between
```

Comment:
On the Java buildpacks, there are a couple of places where we have to handle the Tiny stack specifically. I believe it's presently set up to look at the stack id to determine if we are running on Tiny. If we use the same stack id for all three variants how would one go about detecting when you are running on the tiny stack? Is there other metadata which can be used to determine that?

Reply:
Maybe we can detour into why the buildpacks have that code first. Where are the places that the buildpack relies specifically on the tiny stack ID?

Reply:
The one off the top of my head is Tomcat (and now Tomee). Those servers have a similar shell script that runs to launch the process, so process types for full/base use the script. There's no shell on tiny, so we do some similar work to what the script does in the buildpack and generate a process type that can start the script w/out a shell. It's not as full-featured as the script, but it does make it work.

I also know that we're doing some stuff in the Rust buildpack to handle Tiny specifically as well. The tiny stack is missing a library that Rust compiled apps require, so we're doing a static compile with musl instead of glibc to ensure the binary will run on Tiny.

I can check for other instances tomorrow.

Reply:
Emily mentioned one other case, https://github.com/paketo-buildpacks/rfcs/pull/172#discussion_r848728085, This is for native-image builds with Java. The root problem is similar to Rust. There is library not in Tiny that causes the default build to fail. We add a flag to the build arguments for Tiny which causes the native image to be built in a way that does not require that library. 

I would have to go back and double-check if this is still an issue. The buildpack is 100% still setting this flag, but I haven't checked to see if the present Tiny stack still requires this and don't have notes on exactly what breaks w/out this flag being set.

So in summary:

1. Tomcat & Tomee & Procfile -> look if it's tiny to take action to deal with their being no shell in Tiny.
2. Rust Cargo -> adjusts build arguments to accommodate the fact that tiny does not contain [libgcc1 package](https://packages.ubuntu.com/bionic-updates/libgcc1). Rust code compiled against glibc requires this library.
3. Native Image -> requires setting an alternative flag on build